sys/ztimer/Makefile.dep

```diff
@@ -38,6 +38,12 @@ ifneq (,$(filter xtimer_on_ztimer,$(USEMODULE)))
   PSEUDOMODULES += xtimer_on_ztimer
 endif
 
+# make evtimer use ztimer_msec as low level timer
+ifneq (,$(filter evtimer_on_ztimer,$(USEMODULE)))
+  USEMODULE += ztimer_msec
+  PSEUDOMODULES += evtimer_on_ztimer
+endif
+
```

Comment:
This should be moved next to where the other `evtimer` dependencies are handled.

Reply:
Same question: Also move the `xtimer_on_ztimer` dependency handling to the place where `xtimer` dependencies get handled?

Reply:
Hmm. I'm not so sure anymore. There is also some logic to having the `ztimer` related stuff (even if only indirect) here.

But I agree that either both `xtimer_on_ztimer` and `evtimer_on_ztimer` should be here, or both should be moved to `xtimer` or `evtimer`, respectively.

Reply:
After I slept over it, I would prefer to have the whole dependency resolution for `xtimer` resp. `evtimer` at one place. Currently, for `xtimer` one half can be found in `Makefile.dep` and the other half at `sys/ztimer/Makefile.dep`. (Maybe @kaspar030 has a good reason for that?)

I think the best way is to open another PR fixing this for `xtimer`. Do you agree?

Another question, that just came into my mind: Why is most of the `sys/*` dependency resolution handled in the `Makefile.dep` and not in the `sys/Makefile.dep`? Is there a reason for that? I find this a little bit confusing.

Reply:
> Is there a reason for that?

My guess is: In beginning nobody saw a reason to split it up until it finally became a mess. Now that it became obvious that this file should be split out, new PRs add stuff to `sys/Makefile.dep`. But nobody did yet clean up the old stuff.